### ISSUE TITLE
Add option for tooltips

### ISSUE THREAD

**Issue by rpnfan** [UltimateHackingKeyboard/agent#2289]
Add option to allow entering a tooltip in the Agent GUI for mapped keys and as well for macros, to be displayed in Agent. See the forum post https://forum.ultimatehackingkeyboard.com/t/feature-request-allow-adding-notes-to-key-mappings-in-agent/744/6?u=rpnfan

**Comment by kriker97** [comment]
I vote for this feature in the agent.  It would it be nice to have something like that, I tend to forget what the purpose was being for, when I create a macro after certain time. It could work very well to keep our macros in order.

**Comment by ubuntudroid** [comment]
Oh yes, please please please add something like this!!! ❤️  I configure my keyboard only once every few months and when deciding which rarely used mapping to replace with a new one or how to re-order existing mappings such notes would help a ton!

**Comment by mondalaci** [comment]
I'm ready with the specification.

![image](https://github.com/UltimateHackingKeyboard/agent/assets/300402/2086d0fc-5285-421b-95db-497b2e460cfe)

Notice the button leftward of Cancel. It's a toggle button containing the note Font Awesome icon. When toggling it on, the note field appears:

![image](https://github.com/UltimateHackingKeyboard/agent/assets/300402/23ed826a-7cab-4cff-af19-319246c21572)

- Make the note field two rows large by default and enlarge it as the user adds more lines.
- Display a tooltip when hovering over the note button containing "Add a note to this key." vs "Remove note." according to its state.
- Make the toggled/untoggled state of the note button visible.
- Reduce the vertical space between the Modifiers and Secondary role rows so it's the same as between the Scancode and Modifiers rows.

![dde844825204feeeece92f8ad7013e0b5751c88e](https://github.com/UltimateHackingKeyboard/agent/assets/300402/b3992256-82d7-4ead-b644-8360432c27a8)

- Place the triangle marker underneath the rendered key mapping so it won't cover the mapping.
- Rendering the triangle on non-rectangular keys will be tricky. I suggest specifying an (x,y) translation and rotation for the affected keys in the SVGs to be used by Agent.
- When hovering a key that contains a note, show its note in a tooltip.

Related firmware issue: https://github.com/UltimateHackingKeyboard/firmware/issues/788

**Comment by rpnfan** [comment]
Thanks a lot. That looks great :-)

**Comment by ajh3** [comment]
I would also like tooltips on the keymap screen, mainly for macros. My suggestion was going to be tooltips with the full macro _name_, since right now it's truncated to what can be displayed on each key icon. If you don't think this would overcomplicate things, maybe there could be a checkbox that makes the note equal to the macro title? Just a note would be a lot better than nothing, though. Thank you!

**Comment by mondalaci** [comment]
Maybe we can add a general option that'd expand actions, such as macros in tooltips upon hover, but I'm unsure how widely it'd be used as it hasn't been requested yet. In any case, I wouldn't mix the suggested feature with this one.
